### PR TITLE
Refactor middleware and request handling

### PR DESCRIPTION
**Pull request by tyler-sommer**
HTTP handlers no longer receive a Context. Instead, they receive a specialized Request that contains the original Request and a (possibly empty) Session, auth.Session, and authorized context.Context.

I also moved away from decorating handlers and instead using middlewares set during program start. That gains readability, ie: `mux.HandleContext(auth.Handle(m.authHandler, model.RoleUser, m.someAction))` -> `mux.Handle(m.someAction)` and the modules need less dependencies. However, there are some drawbacks that make me want to reconsider. For example, in middleware.go I have specified how auth is handled for the whole app in one place. This is convenient perhaps but difficult to maintain, and be sure it is correct. So I'll probably switch back to decorators, though I haven't decided how that will look just yet.

### CODE REVIEW

File: http/middleware/middleware.go [hunk 1]
```diff
@@ -0,0 +1,100 @@
...
+			s.manager.BeginAuthorization(a, r, w, req.Request)
+			return nil
+		}
+		req.SetAuthorizedContext(ctx)
+	}
+	return next()
+}
+
+// ignoreRequests returns true if the given request should be ignored.
+func (s *Auth) ignoreRequest(req *route.Request) bool {
```
Comment (tyler-sommer) [inline http/middleware/middleware.go]: This function and `roleForRequest` centralize where auth is configured.

File: http/route/route.go [hunk 2]
```diff
@@ -3,62 +3,134 @@ package route
...
+func (r *Request) AuthorizedContext() (context.Context, bool) {
+	return r.authCtx, r.authCtx != nil
+}
+
+// SetSession sets the request's associated session.
+func (r *Request) SetSession(sess *session.Session) {
+	r.session = sess
+}
+
+// SetAuth sets the request's associated authenticated session.
```
Comment (tyler-sommer) [inline http/route/route.go]: I am also not all that happy with these `Set*` methods, but not sure how else to go about it.

File: http/template/funcs.go [hunk 3]
```diff
@@ -70,16 +68,15 @@ func allianceLogoURL(_ stick.Context, args ...stick.Value) stick.Value {
 	return eveapi.ImageURL(eveapi.ImageAllianceLogo, id, width)
 }
 
-func getContext(sctx stick.Context) context.Context {
-	c, ok := sctx.Scope().Get("context")
+func getRequest(sctx stick.Context) *route.Request {
```
Comment (tyler-sommer) [inline http/template/funcs.go]: Also not that happy with this.

File: http/route/route.go (hunk 2 above)
Comment (zachbadgett) [inline http/route/route.go]: Idiomatic go has an issue with putting Get* in getters, not Set* in setters.

https://golang.org/doc/effective_go.html#Getters

Comment (tyler-sommer) [inline http/route/route.go]: Yeah, I meant more the mutability aspect. Not sure if I gain much by making a Request immutable, though.

**Pull request merged**